### ISSUE TITLE
var-spacing mistakenly flags litteral json patch in a `command` task

### ISSUE THREAD

**Issue by VannTen** [ansible-community/ansible-lint#1792]
##### Summary

Ansible

##### Issue Type

- Bug Report

##### Ansible and Ansible Lint details

```console
ansible --version
ansible [core 2.12.1]
  config file = /home/max/dev_com/kubespray/ansible.cfg
  configured module search path = ['/home/max/dev_com/kubespray/library']
  ansible python module location = /usr/lib/python3.10/site-packages/ansible
  ansible collection location = /home/max/.ansible/collections:/usr/share/ansible/collections
  executable location = /usr/bin/ansible
  python version = 3.10.1 (main, Dec 11 2021, 17:22:55) [GCC 11.1.0]
  jinja version = 3.0.3
  libyaml = True

ansible-lint --version
ansible-lint 5.3.1 using ansible 2.12.1

```

- ansible installation method: OS package
- ansible-lint installation method: OS package

##### OS / ENVIRONMENT

Archlinux, kernel `5.15.10-arch1-1 #1 SMP PREEMPT Fri, 17 Dec 2021 11:17:37 +0000`
##### STEPS TO REPRODUCE

Run ansible-lint on this yaml : (full file here : https://github.com/kubernetes-sigs/kubespray/blob/92d60e27f9f13c317677a3cf3951793b7e6f18a2/roles/win_nodes/kubernetes_patch/tasks/main.yml#L24-L30)

```yaml
    - name: Apply nodeselector patch for kube-proxy daemonset
      command: >
        {{ kubectl }}
        patch ds kube-proxy --namespace=kube-system --type=strategic -p
        {"spec":{"template":{"spec":{"nodeSelector":{"{{ kube_proxy_nodeselector }}":"linux"}}}}}
      register: patch_kube_proxy_state
      when: current_kube_proxy_state.stdout | trim | lower != "linux"
```

##### Desired Behaviour
No error

##### Actual Behaviour

ansible-lint -vvvv output :
```
DEBUG    Logging initialized to level 10
DEBUG    Options: Namespace(cache_dir='/home/max/.cache/ansible-lint/3ec9bc', colored=False, configured=True, cwd=PosixPath('/home/max/dev_com/kubespray'), display_relative_path=True, exclude_paths=['.cache', '.git', '.hg', '.svn', '.tox'], lintables=['roles/win_nodes/'], listrules=False, listtags=False, parseable=True, parseable_severity=False, quiet=False, rulesdirs=['/usr/lib/python3.10/site-packages/ansiblelint/rules'], skip_list=['204', 'meta-no-info', 'role-name', 'var-naming'], tags=[], verbosity=4, warn_list=['experimental', 'role-name'], kinds=[{'jinja2': '**/*.j2'}, {'jinja2': '**/*.j2.*'}, {'inventory': '**/inventory/**.yml'}, {'requirements': '**/meta/requirements.yml'}, {'galaxy': '**/galaxy.yml'}, {'reno': '**/releasenotes/*/*.{yaml,yml}'}, {'playbook': '**/playbooks/*.{yml,yaml}'}, {'playbook': '**/*playbook*.{yml,yaml}'}, {'role': '**/roles/*/'}, {'tasks': '**/tasks/**/*.{yaml,yml}'}, {'handlers': '**/handlers/*.{yaml,yml}'}, {'vars': '**/{host_vars,group_vars,vars,defaults}/**/*.{yaml,yml}'}, {'meta': '**/meta/main.{yaml,yml}'}, {'yaml': '.config/molecule/config.{yaml,yml}'}, {'requirements': '**/molecule/*/{collections,requirements}.{yaml,yml}'}, {'yaml': '**/molecule/*/{base,molecule}.{yaml,yml}'}, {'requirements': '**/requirements.yml'}, {'playbook': '**/molecule/*/*.{yaml,yml}'}, {'yaml': '**/{.ansible-lint,.yamllint}'}, {'yaml': '**/*.{yaml,yml}'}, {'yaml': '**/.*.{yaml,yml}'}], mock_modules=[], mock_roles=[], loop_var_prefix=None, var_naming_pattern=None, offline=False, project_dir='.', extra_vars=None, enable_list=[], skip_action_validation=True, rules={}, format='rich', progressive=False, rulesdir=[], use_default_rules=False, config_file='/home/max/dev_com/kubespray/.ansible-lint', version=False)
DEBUG    /home/max/dev_com/kubespray
INFO     Added ANSIBLE_ROLES_PATH=~/.ansible/roles:/usr/share/ansible/roles:/etc/ansible/roles:roles
DEBUG    Loading rules from /usr/lib/python3.10/site-packages/ansiblelint/rules
WARNING  Loading custom .yamllint config file, this extends our internal yamllint config.
DEBUG    Effective yamllint rules used: {'braces': {'level': 'error', 'forbid': False, 'min-spaces-inside': 0, 'max-spaces-inside': 1, 'min-spaces-inside-empty': -1, 'max-spaces-inside-empty': -1}, 'brackets': {'level': 'error', 'forbid': False, 'min-spaces-inside': 0, 'max-spaces-inside': 1, 'min-spaces-inside-empty': -1, 'max-spaces-inside-empty': -1}, 'colons': {'level': 'error', 'max-spaces-before': 0, 'max-spaces-after': 1}, 'commas': {'level': 'error', 'max-spaces-before': 0, 'min-spaces-after': 1, 'max-spaces-after': 1}, 'comments': {'level': 'warning', 'require-starting-space': True, 'ignore-shebangs': True, 'min-spaces-from-content': 2}, 'comments-indentation': {'level': 'warning'}, 'document-end': False, 'document-start': {'level': 'warning', 'present': True}, 'empty-lines': {'level': 'error', 'max': 2, 'max-start': 0, 'max-end': 0}, 'empty-values': False, 'hyphens': {'level': 'error', 'max-spaces-after': 1}, 'indentation': {'level': 'error', 'spaces': 2, 'indent-sequences': 'consistent', 'check-multi-line-strings': False}, 'key-duplicates': {'level': 'error'}, 'key-ordering': False, 'line-length': False, 'new-line-at-end-of-file': False, 'new-lines': {'level': 'error', 'type': 'unix'}, 'octal-values': False, 'quoted-strings': False, 'trailing-spaces': {'level': 'error'}, 'truthy': False}
INFO     Added ANSIBLE_ROLES_PATH=~/.ansible/roles:/usr/share/ansible/roles:/etc/ansible/roles:roles:roles
INFO     Discovered files to lint using: git ls-files --cached --others --exclude-standard -z
INFO     Excluded removed files using: git ls-files --deleted -z
DEBUG    Examining roles/win_nodes of type role
DEBUG    Examining roles/win_nodes/kubernetes_patch/tasks/main.yml of type tasks
DEBUG    Examining roles/win_nodes/kubernetes_patch/defaults/main.yml of type vars
WARNING  Listing 1 violation(s) that are fatal
roles/win_nodes/kubernetes_patch/tasks/main.yml:24: var-spacing Variables should have spaces before and after: }}":"linux"}}}}}

You can skip specific rules or tags by adding them to your configuration file:
# .ansible-lint
warn_list:  # or 'skip_list' to silence them completely
  - var-spacing  # Variables should have spaces before and after:  {{ var_name }}

Finished with 1 failure(s), 0 warning(s) on 3 files.
```

[minimum complete verifiable example]: http://stackoverflow.com/help/mcve

**Comment by ssbarnea** [comment]
I will close this because it can easily be avoided with a noqa and also because I think it would be very hard to fix the regular expression that we use for finding these.

Don't be fooled by closed status, if you can produce a pull-request to improve it, I will the first to review it.
